### PR TITLE
boards/stk3200: add support

### PR DESCRIPTION
**Pull request by basilfx**
### Contribution description
This PR add supports for the Silicon Labs STK3200, which I already PR'd before in #8295 and #8832. The files are similar to the SLTB001a and STK3600, STK3700 and SLSTK3401a already supported. This is a Cortex M0+ based board, with a low-power memory LCD that is supported by U8g2. The board has only 32KiB of flash, so most tests have been ignored.

### Issues/PRs references
#8832
#8295

### CODE REVIEW

**Comment by dylad** [comment]
@basilfx UART TX seems to work but I cannot interact with the shell, is RX working on your side ?

File: boards/stk3200/include/board.h [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+/** @} */
+
+/**
+ * @name    Board controller configuration
+ *
+ * Define the GPIO pin to enable the BC, to allow serial communication
+ * via the USB port.
+ * @{
+ */
+#define BC_PIN              GPIO_PIN(PA, 9)
```
Comment (dylad) [inline boards/stk3200/include/board.h]:
```suggestion
#define BC_PIN              GPIO_PIN(PD, 7)
```

**Comment by basilfx** [comment]
Shell is working for me (without the fix).

image

I have BRD 2010A Rev. B00.

File: boards/stk3200/include/board.h (hunk 1 above)
Comment (basilfx) [reply boards/stk3200/include/board.h]: That does make sense, according to the [manual](https://www.silabs.com/documents/public/user-guides/efm32zg-stk3200-ug.pdf). But it's strange that it does work for me. I'll try as well.

Comment (basilfx) [reply boards/stk3200/include/board.h]: It seems that this breaks RX for me.

Comment (basilfx) [reply boards/stk3200/include/board.h]: The schema is [contradicting](https://www.silabs.com/documents/public/schematic-files/EFM32ZG_BRD2010A_schematic.pdf)

**Comment by dylad** [comment]
This is so weird.
I have BRD 2010A Rev B00 too
PCB rev: PCB2010 REVB02

**Comment by dylad** [comment]
run compile_and_test_for_board.py script all night, here are the results:

```
- [tests/cpp11_mutex](tests/cpp11_mutex/test.failed)
- [tests/events](tests/events/test.failed)
- [tests/periph_flashpage](tests/periph_flashpage/test.failed)
- [tests/pkg_hacl](tests/pkg_hacl/test.failed)
- [tests/pkg_ubasic](tests/pkg_ubasic/test.failed)
- [tests/pthread_barrier](tests/pthread_barrier/test.failed)
- [tests/pthread_cleanup](tests/pthread_cleanup/test.failed)
- [tests/pthread_cooperation](tests/pthread_cooperation/test.failed)
- [tests/pthread_tls](tests/pthread_tls/test.failed)
- [tests/shell](tests/shell/test.failed)
- [tests/stdin](tests/stdin/test.failed)
- [tests/sys_arduino](tests/sys_arduino/test.failed)
- [tests/xtimer_msg](tests/xtimer_msg/test.failed)
- [tests/xtimer_periodic_wakeup](tests/xtimer_periodic_wakeup/test.failed)
```

File: boards/stk3200/doc.txt [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+| Flash           | 32.0KB                                               |
+| EEPROM          | no                                                   |
+| Frequency       | up to 24 MHz                                         |
+| FPU             | no                                                   |
+| MPU             | no                                                   |
+| DMA             | 12 channels                                          |
+| Timers          | 2x 16-bit + 1x 16-bit (low power)                    |
+| ADCs            | 12-bit ADC                                           |
+| UARTs           | 2x USART, 1x LEUART                                  |
+| SPIs            | 2x USART                                             |
```
Comment (benpicco) [inline boards/stk3200/doc.txt]: USART?

**Comment by benpicco** [comment]
@basilfx want to give this a rebase?`

**Comment by benpicco** [comment]
This one is pretty much good to go.
Just needs a rebase :smiley: 

Btw.: Where did you get the vendor files? I'm looking for the ones for `EFR32MG1B` :innocent: 

Since the last two commits cause a lot of conflicts, just drop those and run

    ./dist/tools/insufficient_memory/add_insufficient_memory_board.sh stk3200

**Review by benpicco — APPROVED** [review]
Rebased, added Kconfig and fixed the merge conflict.
The LETIMER fix was merged independent from this, so this should be good :innocent:

**Pull request merged**